src/Dapr.Actors/DaprGrpcInteractor.cs

```diff
@@ -0,0 +1,364 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+namespace Dapr.Actors
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.IO;
+    using System.Linq;
+    using System.Net.Http;
+    using System.Text;
+    using System.Text.Json;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Dapr.Actors.Communication;
+    using Dapr.Actors.Resources;
+    using System.Xml;
+    using Autogenerated = Dapr.Client.Autogen.Grpc.v1;
+    using Grpc.Core;
+    using Google.Protobuf;
+    using Dapr.Actors.Runtime;
+    using Grpc.Net.Client;
+
+    /// <summary>
+    /// Class to interact with Dapr runtime over grpc.
+    /// </summary>
+    internal class DaprGrpcInteractor : IDaprInteractor
+    {
+        private readonly JsonSerializerOptions jsonSerializerOptions = JsonSerializerDefaults.Web;
+        private bool disposed;
+        private string daprApiToken;
+        private readonly Autogenerated.Dapr.DaprClient client;
+        internal Autogenerated.Dapr.DaprClient Client => client;
+        private readonly GrpcChannel channel;
+
+        private const string EXCEPTION_HEADER_TAG = "b:KeyValueOfstringbase64Binary";
+
+        public DaprGrpcInteractor(
+            GrpcChannel channel,
+            Autogenerated.Dapr.DaprClient inner,
+            string apiToken)
+        {
+            this.channel = channel;
+            this.client = inner;
+            this.daprApiToken = apiToken;
+        }
+
+        public async Task<string> GetStateAsync(string actorType, string actorId, string keyName, CancellationToken cancellationToken = default)
+        {
+            var request = new Autogenerated.GetActorStateRequest()
+            {
+                ActorId = actorId,
+                ActorType = actorType,
+                Key = keyName,
+            };
+            var options = CreateCallOptions(cancellationToken);
+
+            Autogenerated.GetActorStateResponse response = new Autogenerated.GetActorStateResponse();
+            try
+            {
+                response = await client.GetActorStateAsync(request, options);
+            }
+            catch (RpcException ex)
+            {
+                throw new DaprApiException("GetActorState operation failed: the Dapr endpoint indicated a failure. See InnerException for details.", ex);
+            }
+            return response.Data.ToStringUtf8();
+        }
+
+        public Task SaveStateTransactionallyAsync(string actorType, string actorId, string data, CancellationToken cancellationToken = default)
+        {
+            return null;
+        }
```

Comment:
Why did we decide to do this and create a new method instead of just using the gRPC client here?